sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -751,7 +751,8 @@ object IntervalUtils {
       secs: Decimal): CalendarInterval = {
     val totalMonths = Math.addExact(months, Math.multiplyExact(years, MONTHS_PER_YEAR))
     val totalDays = Math.addExact(days, Math.multiplyExact(weeks, DAYS_PER_WEEK))
-    var micros = (secs * Decimal(MICROS_PER_SECOND)).toLong
+    assert(secs.scale == 6, "Seconds fractional must have 6 digits for microseconds")
```

Comment:
shall we check the precision as well?

Reply:
We can check it but `precision` value is not important for this code...or I am wrong?

Reply:
So, `Decimal` guarantees that `precision` >= `scale` (@Ngone51 correct?). That's enough for the code.

Reply:
Technically, I think we should say `DecimalType`  guarantees `precision` >= `scale`. But since any `Decimal` is directly or indirectly bounded with a `DecimalType`. Therefore, `Decimal` also guarantees `precision` >= `scale`.

Reply:
I just want to be consistent with https://github.com/apache/spark/pull/28886/files#diff-b83497f7bc11578a0b63a814a2a30f48R2003

Reply:
I added assert there to guarantee that Int will not overflow. What would you like to assert here?

Reply:
for sanity check? The `precision` must be `<= 8` here, right?

Reply:
Not anymore, please, take a look at https://github.com/apache/spark/pull/28873